launch_ros/launch_ros/actions/node.py

```diff
@@ -135,8 +147,18 @@ def __init__(
         # The substitutions will get expanded when the action is executed.
         cmd += ['--ros-args']  # Prepend ros specific arguments with --ros-args flag
         if node_name is not None:
+            warnings.warn("The parameter 'node_name' is deprecated, use 'name' instead")
+            if name is None:
```

Comment:
Passing both `name` and `node_name` should be detected as an error IMO. I would add an else case.

Reply:
Raising an exception if this happens: fa4ab9f